samples/go/csv/csv-import/importer.go

```diff
@@ -63,7 +64,7 @@ func main() {
 	switch {
 	case flag.NArg() == 0:
 		err = errors.New("Maybe you put options after the dataset?")
-	case flag.NArg() == 1 && *path == "":
+	case flag.NArg() == 1 && *path == "" && !*detectColumnTypes:
```

Comment:
Because if I am just detecting column types, I don't actually need the dataset arg.